modules/gis/src/lib/geojson-to-binary.js

```diff
@@ -391,14 +391,14 @@ function fillNumericProperties(object, properties, index, length) {
 }
 
 // Keep string properties in object
-// Note: this mutates the properties object
 function keepStringProperties(properties, numericKeys) {
-  for (const key in properties) {
+  const props = {...properties};
+  for (const key in props) {
```

Comment:
Alternatively we could create an empty object and just add the props we want.

Reply:
What do you prefer? I can change it if you want, it sounds better for me as it makes less operations.

Reply:
I'm a fan of filling an empty object. But still, note that for deeply-nested objects the new and old objects are still linked, and just filling an empty object doesn't perform deep cloning.

```js
var x = {a: {b: {c: 2}}};
var newObj = {};
newObj.a = x.a;

console.log(newObj.a.b.c);
// 2

x.a.b.c = 3

console.log(newObj.a.b.c);
// 3
```

Reply:
> What do you prefer? I can change it if you want, it sounds better for me as it makes less operations.

Sure. It is hard to predict what will actually be faster in the end with modern JavaScript engines, but it doesn't really matter here, and it makes sense to just add the props we need.

> just filling an empty object doesn't perform deep cloning.

Deep cloning seems expensive. Is there a reason to offer that option here?

Reply:
> Deep cloning seems expensive. Is there a reason to offer that option here?

I don't think there necessarily is, but it's important to keep in mind that the objects are still linked. When the input object is mutated, it's clear that input and output are "linked" and that there's no safety in only changing one object. When the input object isn't mutated, it might not be clear to all users that changing `newFeature.properties.a` is still unsafe if you expect to use `oldFeature` again

Reply:
Yes, this is something @alasarr and I discussed, but as long as a geojson doesn't use to use nested objects for its properties I tried to keep it simpler, but thinking about it again after your explanation, I think that it can give a false "feeling of security" about not mutating the input object.

So, I suggest two solutions. One option would be to let developers clearly be aware that this operation will mutate the input object in the documentation and/or using JSDoc to show it while they are coding (I really didn't see it when I started working with `geojsonToBinary`, it currently is only written in the inner function that mutates the properties object), so they can deep clone, freeze or so their geojson objects before using it in case they need it.
The other one, would be adding a new flag in `options` for `geojsonToBinary()` that would make the deep clone for you, having this option set to false by default, so in the documentation of this option we could make them aware of the mutation and let them get rid of it easily, making them aware that this flag could impact the performance.

Personally, I think that the first option would be enough.

Reply:
I think we are overworking this. If I understand it, some properties can't be converted to binary, for those we return "classic" arrays, and those arrays can share nested objects with the original JSON? If so just mention that in the docs...

Reply:
Yes this is more or less the story, just adding that the properties object from the original JSON have its numeric props deleted as well.
So, to confirm, let's change the behavior from deleting items from the original to adding to a new object and add a mention in the docs about this shared nested objects, agree?

Reply:
I've uploaded the proposal I made, please take a look to it, thanks!